src/TraceManager.py

```diff
@@ -0,0 +1,62 @@
+"""
+<Program Name>
+  TraceManager
+
+<Started>
+  July 2019
+
+<Author>
+  Junjie Ge
+
+<Purpose>
+  Manages the trace identified by the producer (posix_omni_parser)
+
+"""
+
+class TraceManager:
+  def __init__(self):
+      self.syscall_objects = []
+      self.trace = []
+      self.mutators = []
+
+  def register_mutator(self, mutator):
+      self.mutators.append({'name': mutator, 'index': 0})
+  
+  def pop_front(self):
+      self.syscall_objects.pop(0)
+      self.trace.pop(0)
+      for mutator in self.mutators:
+          mutator['index'] -= 1
+
+  def next_syscall(self, calling_mutator):
+      mutator_index = -1
+      for i in range(len(self.mutators)):
+          if self.mutators[i]['name'] == calling_mutator:
+              mutator_index = i
+              break
+      if mutator_index == -1:
+          print('----')
```

Comment:
We should not print here.  This could be logging statements at an appropriate log level

Reply:
Done